packages/ui/src/components/Modal/Disclosure.tsx

```diff
@@ -0,0 +1,46 @@
+import type { PropsWithRef } from 'react'
+import { cloneElement, createRef, isValidElement, useEffect } from 'react'
+import type { DisclosureProps } from './types'
+
+export const Disclosure = ({
+  disclosure,
+  handleOpen,
+  visible,
+  handleClose,
+  toggle,
+  id,
+}: DisclosureProps) => {
+  const disclosureRef = createRef<HTMLElement>()
+
+  useEffect(() => {
+    const element = disclosureRef.current
+    const handleClick = () => {
+      handleOpen()
+    }
+
+    element?.addEventListener('click', handleClick)
+
+    return () => {
+      element?.removeEventListener('click', handleClick)
```

Comment:
So we can directly use `handleOpen` instead of having another function?
```suggestion
    element?.addEventListener('click', handleOpen)

    return () => {
      element?.removeEventListener('click', handleOpen)
```